src/clinical/clinical-service.ts

```diff
@@ -324,14 +334,69 @@ export const getDonorSubmissionErrorUpdates = async (
     submitterDonorIds: [],
   };
 
-  await donorDao.findByPaginatedProgramId(programId, errorQuery).then(donorData => {
-    validDonors = donorData.donors
-      .filter(donor => donor.schemaMetadata.isValid)
-      .map(({ donorId }) => Number(donorId));
-  });
+  const donorData = (await donorDao.findByPaginatedProgramId(programId, errorQuery)).donors;
+
+  const validDonorIds = donorData
+    .filter(donor => donor.schemaMetadata.isValid)
+    .map(({ donorId }) => Number(donorId));
```

Comment:
`donorId` is a number already

Reply:
True, removed
`donorId` is potentially `undefined` but either way .map is just returning something falsey, no impact on `validDonorIds` usage